rs/registry/canister/src/mutations/routing_table.rs

```diff
@@ -31,18 +36,105 @@ impl std::fmt::Display for GetSubnetForCanisterError {
     }
 }
 
+/// Complexity O(n)
+// TODO after migration runs in registry_lifecycle.rs, make this function private to this module again.
+pub(crate) fn mutations_for_canister_ranges(
+    old_rt: &RoutingTable,
+    new_rt: &RoutingTable,
+) -> Vec<RegistryMutation> {
+    // Helper functions
+    let range_key = |key: (CanisterId, SubnetId)| -> Vec<u8> {
+        make_canister_range_key(key.0, key.1).as_bytes().to_vec()
+    };
+
+    let create_entry = |key: (CanisterId, SubnetId), range_end: CanisterId| -> Vec<u8> {
+        pb::routing_table::Entry {
+            range: Some(pb::CanisterIdRange {
+                start_canister_id: Some(key.0.into()),
+                end_canister_id: Some(range_end.into()),
+            }),
+            subnet_id: Some(pb_subnet_id(key.1)),
```

Comment:
Why repeat the subnet id which is already represented in the key?

Reply:
Much easier to process the output - clients won't have to parse subnet strings and reconstruct the records but can rely on the protobuf encoding, which is more robust.

Reply:
Sure, but then how about dropping the subnet ID from the key?